### ISSUE TITLE
Not getting any events for connectionUpdated/purchaseUpdated/purchaseError subscriptions. Any ideas?

### ISSUE THREAD

**Issue by mpodolyako** [dooboolab/flutter_inapp_purchase#127]
- Start the example app
- Click on 'Connect Billin' button - no logs about connection state appeared
- Click on 'Get Items' button - but it's possible to get list of items

### Version of flutter_inapp_purchase
2.0.4; flutter version 1.9.1+hotfix.6

### Platforms you faced the error (IOS or Android or both?)
iOS 12.4.3

### Expected behavior
After clicking on 'Connect Billin' I expect to see 'connected ...' log message

### Actual behavior
After clicking on 'Connect Billin' button I see only 'Connect Billing Button Pressed!', but no log message about connection state. Nevertheless, if I click on 'Get Items' button, I can see products details on the device's screen, which means the connection was actually successful.
Subscriptions for some reason don't work. Does anyone has any ideas?..

### Tested environment (Emulator? Real Device?)
real device

**Comment by letiagoalves** [comment]
I am having the same problem. In my case case it is a race condition. The broadcast stream is being called before the listener is attached. I believe that each stream should hold the last emitted value and replay it for new subscriptions.

**Comment by hyochan** [comment]
There was a PR in #126 which looks related. Could you guys try `2.0.5` and come back?

**Comment by mpodolyako** [comment]
@hyochan, #126 looks for android only (according to changed file AndroidInappPurchasePlugin.java)

@letiagoalves 

> I am having the same problem. In my case case it is a race condition. The broadcast stream is being called before the listener is attached. I believe that each stream should hold the last emitted value and replay it for new subscriptions.

looks true, but my problem is I'm not getting events not only for the connectionUpdated stream, but also for the purchaseUpdated stream as well. 

To clarify that once agian: I've changed productIds in the example app according to that I have in my app. Also, I've changed iOS bundleId to be able to buy some product under my iOS sanbox account.
No other code changes. Then, I'm running the example app. 
1. Click 'Connect Billin'
2. 'connected' log message is missing
3. Click 'Get Items' - i'm able to get list of my items
4. 'Buy Item'
5. I see 'Purchase Started !!' in logs. iOS asking me for a password and confirmation to buy the product. 
6. iOS shows me the product is bought, but I see no 'purchase-updated: ...' log message, so I'm not receiving any events for the purchaseUpdated stream.

That most probably some issue with my env (flutter, xcode, ios, etc...). Just asking whether anyone faced similar issue, because I don't know what else to check.
